samples/Event_Jsvm/app/application.cpp

```diff
@@ -1,41 +1,109 @@
 #include <SmingCore.h>
-#include <Jsvm.h>
-#include <vm_functions.h>
+#include <Jerryscript.h>
 
 namespace
 {
-
-Jsvm jsVm;
-Timer tempTimer;
+JS::VirtualMachine vm;
+SimpleTimer timer;
+HashMap<String, JS::Callable::List> events;
 
 IMPORT_FSTR(main_snap, PROJECT_DIR "/out/jerryscript/main.js.snap")
 
+namespace JS
+{
+using namespace Jerryscript;
+
+/**
+ * @brief Function to register event listeners
+ * @code {.javascript}
+ * 
+ * event = {
+ *    name => "TEMP_CHANGE",
+ *    params = {
+ *    },
+ *    "origin" => "The\Creator\Of\The\Event" 
+ * };
+ * 
+ * addEventListener("TEMP_CHANGE", function(event) {
+ *      console.log("Got Event" + event.name);
+ * });
+ * 
+ * @endcode
+ * 
+ */
+Value addEventListener(const CallInfo& callInfo, Value& eventName, Callable& function)
+{
+	if(!eventName.isString() || !function.isCallable()) {
+		return ArgumentError(__FUNCTION__);
+	}
+
+	events[eventName].add(function);
+	return true;
+}
+
+} // namespace JS
+
+JS_DEFINE_FUNCTION(addEventListener, 2)
+
+/*
+ * Dispatch an event to all registered listeners
+ *
+ * Event = {
+ * 		name: "eventName"
+ * 		params: {
+ * 			"property1": value1,
+ * 			"property2": value2,
+ * 			...
+ * 		}
+ * }
+ */
+bool triggerEvent(const String& name, const JS::Object& params)
+{
+	if(!events.contains(name)) {
+		debug_e("%s: Unknown event '%s'", __FUNCTION__, name.c_str());
+		return false;
+	}
+
+	// Build the event object...
+	JS::Object event;
+	event["name"] = name;
+	event["params"] = params;
+
+	auto realm = JS::global();
+	for(auto& listener : events[name]) {
+		listener.call(realm, event);
+	}
+
+	return true;
+}
+
 void startJsvm()
 {
 	/*
 	 * This is how we register a new function in JavaScript
 	 * that will communicate directly with our C/C++ code.
 	 */
-	jsVm.registerFunction("addEventListener", addEventListener);
+	vm.registerFunction("addEventListener", addEventListener);
```

Comment:
As above, this could translate into `JS::global().registerFunction(...)`

Reply:
Let's have it as suggested.